.github/workflows/ci.yaml

```diff
@@ -0,0 +1,71 @@
+# Copyright 2024 NVIDIA CORPORATION
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: CI
+
+on:
+  pull_request:
+    types:
+      - opened
+      - synchronize
+    branches:
+      # - main
+      # - release-*
+      - drivervalidation
+  push:
+    branches:
+      # - main
+      # - release-*
+      - drivervalidation
+jobs:
+  e2e-tests-nvidiadriver:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Check out code
+      uses: actions/checkout@v4
+        
+    - name: Set up Holodeck
+      uses: NVIDIA/holodeck@main
+      env:
+        AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+        AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+        AWS_SSH_KEY: ${{ secrets.AWS_SSH_KEY }}
+        AWS_SESSION_TOKEN: ${{ secrets.AWS_SESSION_TOKEN }}
+      with:
+        aws_access_key_id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+        aws_secret_access_key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+        aws_ssh_key: ${{ secrets.AWS_SSH_KEY }}
+        holodeck_config: "tests/holodeck.yaml"
+        
+    - name: Get public dns name
+      id: get_public_dns_name
+      uses: mikefarah/yq@master
+      with:
+        cmd: yq '.status.properties[] | select(.name == "public-dns-name") | .value' /github/workspace/.cache/holodeck.yaml  
+
+    - name:  Set and Calculate test vars
+      run: |
+        echo "instance_hostname=ubuntu@${{ steps.get_public_dns_name.outputs.result }}" >> $GITHUB_ENV
```

Comment:
The one thing missing from this PR is that we are not testing the driver images pushed to ghcr.io. Right now, we are installing the GPU Operator with default options, which means the driver is getting pulled from nvcr.io.

Can you update this so that we test driver container images from ghcr.io built for the commit sha? 

To get the commit sha: https://github.com/NVIDIA/gpu-driver-container/blob/main/.github/workflows/image.yaml#L70

The images to test are pushed to `ghcr.io/nvidia/gpu-driver-container:<commit-short-sha>-<driver-version>-ubuntu22.04`. So we need to install the operator and set `driver.repository=ghcr.io/nvidia` and `driver.version=<commit-short-sha>-<driver-version>`

The one caveat of this is that the e2e workflow needs to run after the images are built, otherwise we will get image pull errors. As I suggested in another comment, let's limit this workflow to manual invocations, and we will only run it once the images have been pushed to ghcr.io.